File: src/d1hashmaps/Java/Bucket.java
import java.util.ArrayList;
import java.util.List;

public class Bucket {
    private List<Pair> pairs;

    public Bucket() {
        this.pairs = new ArrayList<>();
    }

    public int get(int key) {
        for (Pair pair : pairs) {
            if (pair.key == key) {
                return pair.value;
            }
        }
        return -1; // Return -1 if the key does not exist
    }

    public void update(int key, int value) {
        for (Pair pair : pairs) {
            if (pair.key == key) {
                pair.value = value;
                return;
            }
        }
        pairs.add(new Pair(key, value));
    }

    public void remove(int key) {
        pairs.removeIf(pair -> pair.key == key);
    }

    public List<Pair> getPairs() {
        return pairs;
    }
}

File: src/d1hashmaps/Python/Bucket.py
from Pair import Pair


class Bucket:
    def __init__(self):
        self.pairs = []

    def get(self, key):
        for pair in self.pairs:
            if pair.key == key:
                return pair.value
        return -1  # Return -1 if the key does not exist

    def update(self, key, value):
        for pair in self.pairs:
            if pair.key == key:
                pair.value = value
                return
        self.pairs.append(Pair(key, value))

    def remove(self, key):
        self.pairs = [pair for pair in self.pairs if pair.key != key]
